### PR TITLE
Adding .Net SDK support for Virtual Cluster operations

### PR DESCRIPTION
**Pull request by milostod**
This change adds support for Virtual Cluster operations. Corresponding Swagger spec changes are already committed, see PR: https://github.com/Azure/azure-rest-api-specs/pull/5248
Note: Virtual Cluster doesn't support create operation. It is created implicitly when first Managed Instance is created in some VirtualNetwork/Subnet.

### CODE REVIEW

File: src/SDKs/SqlManagement/Sql.Tests/Sql.Tests.csproj [hunk 1]
```diff
@@ -52,4 +52,9 @@
     <Folder Include="SessionRecords\Sql.Tests.ManagedInstanceVulnerabilityAssessmentScenarioTests\" />
     <Folder Include="SessionRecords\Sql.Tests.ServerVulnerabilityAssessmentScenarioTests\" />
   </ItemGroup>
+  <ItemGroup>
+    <Compile Update="VirtualClusterCrudScenarioTests.cs">
```
Comment (dsgouda) [inline src/SDKs/SqlManagement/Sql.Tests/Sql.Tests.csproj]: Was this intended?

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks good for the most part

File: src/SDKs/SqlManagement/Sql.Tests/Sql.Tests.csproj (hunk 1 above)
Comment (milostod) [reply src/SDKs/SqlManagement/Sql.Tests/Sql.Tests.csproj]: No, thanks for noticing. I removed this.

**Review by dsgouda — APPROVED** [review]
LGTM

File: src/SDKs/SqlManagement/Management.Sql/Generated/SqlManagementClient.cs [hunk 2]
```diff
@@ -442,19 +447,6 @@ public partial class SqlManagementClient : ServiceClient<SqlManagementClient>, I
...
 
-        /// <summary>
-        /// Initializes a new instance of the SqlManagementClient class.
-        /// </summary>
-        /// <param name='httpClient'>
-        /// HttpClient to be used
-        /// </param>
-        /// <param name='disposeHttpClient'>
-        /// True: will dispose the provided httpClient on calling SqlManagementClient.Dispose(). False: will not dispose provided httpClient</param>
-        protected SqlManagementClient(HttpClient httpClient, bool disposeHttpClient) : base(httpClient, disposeHttpClient)
```
Comment (milostod) [inline src/SDKs/SqlManagement/Management.Sql/Generated/SqlManagementClient.cs]: These two constructors were automatically removed. I'm not sure what triggered the change.

**Pull request merged**